### PR TITLE
Add beta migration values examples and README

### PR DESCRIPTION
**Pull request by dbkegley**

### CODE REVIEW

File: examples/connect/beta-migration/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+These example values files define the default set of `bionic` images in the
+`launcher.customRuntimeYaml` section, but each image's "matching" has been set to `exact`.
+The `launcher.additionalRuntimeImages` contains the _new_ default set of `jammy` images,
+however the `jammy` images _do not_ define a matching strategy.
+
+Any new content that is building for the first time, or existing content that is re-building,
+will attempt to match against the set of `jammy` images. Content that has already been
+built with a `bionic` image can continue to _execute_ with that image until it rebuilds. If an existing
+piece of content needs to be rebuilt and requires `bionic` for some reason, the Publisher
+must _explcitly_ define a default execution enviornment for that content through the content-settings
```
Comment (colearendt) [inline examples/connect/beta-migration/README.md:25]:
```suggestion
must _explcitly_ define a default execution environment for that content through the content-settings
```

File: examples/connect/beta-migration/values-base.yaml [hunk 2]
```diff
@@ -0,0 +1,202 @@
+launcher:
```
Comment (colearendt) [inline examples/connect/beta-migration/values-base.yaml:1]: Why don't we set the config bits at the top, leave the `jammy` images as the "default" from the chart, and add the `additional` images as the bionic ones that are `exact` matches? Wouldn't that simplify this file a bit?

File: examples/connect/beta-migration/values-pro.yaml [hunk 3]
```diff
@@ -0,0 +1,261 @@
+launcher:
```
Comment (colearendt) [inline examples/connect/beta-migration/values-pro.yaml:1]: Same concern on over-complicating

File: examples/connect/beta-migration/README.md (hunk 1 above)
Comment (colearendt) [inline examples/connect/beta-migration/README.md:25]:
```suggestion
must _explicitly_ define a default execution environment for that content through the content-settings
```

Comment (colearendt) [inline examples/connect/beta-migration/README.md:25]:
```suggestion
must _explicitly_ define a default execution environment for that content through the content-settings
```

File: charts/rstudio-connect/NEWS.md [hunk 4]
```diff
@@ -8,6 +8,9 @@
     `launcher.defaultInitContainer.tagPrefix=bionic-` (content-init),
     and modify the set of content images defined by `launcher.customRuntimeYaml`
     to use `bionic` instead of `ubuntu2204`
+  - Off-Host Execution Beta users who are currently evaluating this feature set can use the
```
Comment (colearendt) [inline charts/rstudio-connect/NEWS.md:11]: Maybe add a "BREAKING" note to highlight?

**Review by colearendt — COMMENTED** [review]
My only real concerns:
- I think some of this feels very explanatory, but perhaps "too" verbose

I would lean towards:
- a simple note in the main `README` that says something to the effect of "BREAKING Beta to GA" -> Make sure you read XYZ doc on migration- with link
- Simply explain: The off-host execution beta moving to GA also moves your content from Bionic to Jammy by default. This will cause all content to get rebuilt
- To avoid the breaking change, either:
    - make sure that your previously used images are specified in `additionalRuntimeImages` or `customRuntimeImages`
     - If you are currently using `customRuntimeImages: 'base'` (the default), use this values file with `helm upgrade --install -f URL-WITH-FIX`
     - If you are currently using `customRuntimeImages: 'pro'`, use other values file with `helm upgrade --install -f URL-WITH-FIX`
     
 Basically, my suspicion is that folks will want the decision tree with steps to resolve more than they will care about the intricate explanation, so we only need to give them a small bit of info (migrating bionic to jammy, what it will do, how to avoid / fix / prevent)
 
 I think saving the explanation is good, I would just prioritize the "linear path to happiness" 😄

File: examples/connect/beta-migration/values-base.yaml (hunk 2 above)
Comment (dbkegley) [reply examples/connect/beta-migration/values-base.yaml:1]: Good call. Done!

File: examples/connect/beta-migration/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+> Note: All images used by existing content should be specified in either
+`launcher.customRuntimeImages` or `launcher.additionalRuntimeImages`. If content
+requires both bionic and jammy images, then you must use R source packages.
+Binary packages will not work when multiple OS distributions are used.
+
+1. If you are currently using `customRuntimeImages: 'base'` (the default),
+apply these changes with:
+
+```
+helm upgrade --install -f https://raw.githubusercontent.com/rstudio/helm/main/examples/connect/beta-migration/values-base.yaml
```
Comment (colearendt) [inline examples/connect/beta-migration/README.md:19]: Sorry, I used pseudocode on these. Still need a release name and chart reference 🙈 Happy to drop it if you don't think this is worth it. 
```suggestion
helm upgrade --install my-release rstudio/rstudio-connect --devel -f https://raw.githubusercontent.com/rstudio/helm/main/examples/connect/beta-migration/values-base.yaml
```

File: examples/connect/beta-migration/README.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+2. If you are currently using `customRuntimeImages: 'pro'`, apply these changes with:
+
+```
+helm upgrade --install -f https://raw.githubusercontent.com/rstudio/helm/main/examples/connect/beta-migration/values-pro.yaml
```
Comment (colearendt) [inline examples/connect/beta-migration/README.md:25]:
```suggestion
helm upgrade --install my-release rstudio/rstudio-connect --devel -f https://raw.githubusercontent.com/rstudio/helm/main/examples/connect/beta-migration/values-pro.yaml
```

File: examples/connect/beta-migration/values-base.yaml [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+          - path: /opt/R/4.1.3/bin/R
+            version: 4.1.3
+      quarto:
+        installations:
+          - path: /opt/quarto/bin/quarto
+            version: 1.0.35
+
+# -- A nested map of maps that generates the rstudio-connect.gcfg file
+# @default -- [RStudio Connect Configuration Reference](https://docs.rstudio.com/connect/admin/appendix/configuration/)
+config:
```
Comment (colearendt) [inline examples/connect/beta-migration/values-base.yaml:132]: I would also probably move this up, but not a big deal

**Review by colearendt — APPROVED** [review]
LGTM! Just a few nits

File: examples/connect/beta-migration/README.md (hunk 5 above)
Comment (dbkegley) [reply examples/connect/beta-migration/README.md]: I went ahead and removed this for now. Since these are choices that must be made during a migration I think it's best if users have to be deliberate about choosing images they want to use.
